sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ContinuousAccessEvaluationTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Tests;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Administration.Tests
+{
+    [NonParallelizable]
+    internal class ContinuousAccessEvaluationTests : ContinuousAccessEvaluationTestsBase
+    {
+        [SetUp]
+        public void Setup()
+        {
+            ChallengeBasedAuthenticationPolicy.ClearCache();
+        }
+
+        [Test]
+        [TestCase(@"Bearer realm="""", authorization_uri=""https://login.microsoftonline.com/common/oauth2/authorize"", error=""insufficient_claims"", claims=""eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ==""", """{"access_token":{"nbf":{"essential":true,"value":"1726077595"},"xms_caeerror":{"value":"10012"}}}""")]
+        public async Task VerifyCaeClaims(string challenge, string expectedClaims)
+        {
+            int callCount = 0;
+
+            MockResponse response = new MockResponse(200);
+
+            MockTransport transport = GetMockTransportWithCaeChallenges(numberOfCaeChallenges: 1, final200response: response);
+
+            var credential = new TokenCredentialStub((r, c) =>
+            {
+                if (callCount == 0)
+                {
+                    // The first challenge should not have any claims.
+                    Assert.IsNull(r.Claims);
+                }
+                else if (callCount == 1)
+                {
+                    Assert.AreEqual(expectedClaims, r.Claims);
+                }
+                Interlocked.Increment(ref callCount);
+                Assert.AreEqual(true, r.IsCaeEnabled);
+
+                return new(callCount.ToString(), DateTimeOffset.Now.AddHours(2));
```

Comment:
Looks like this test would pass if the policy authorized the request with the same token before and after the claims challenge. We should also verify that it uses the new token after the challenge.